packages/block-editor/src/components/block-title/index.js

```diff
@@ -40,5 +53,13 @@ export default function BlockTitle( { clientId } ) {
 		return null;
 	}
 
-	return blockType.title;
+	const { title } = blockType;
+	const label = getBlockLabel( blockType, attributes );
+
+	// Label will often fall back to the title if no label is defined for the
+	// clurrent label context. We do not want "Paragraph: Paragraph".
+	if ( label !== title ) {
+		return `${ title }: ${ truncate( label, { length: 15 } ) }`;
```

Comment:
I don't know if this was discussed but are we certain that "truncate" works well with internationalized strings? Should we rely on CSS truncate instead?

Reply:
It's [aware of unicode chars](https://github.com/lodash/lodash/blob/86a852fe763935bb64c12589df5391fd7d3bb14d/truncate.js#L68-L71), and it seems to handle them well in my (naïve tests).

I've tried truncating a string with Japanese kanji and one with "complex" emoji, and in both cases the truncation happened as expected, with kanji and emojis being treated as single characters.

```js
truncate( 'lorem ipsum dolor sit amet', { length: 10 } );
// "lorem i..."

truncate( 'kanji 僕はばかです', { length: 10 } );
// "kanji 僕..."

truncate( 'emoji 👩‍🍳👡👩‍👩‍👦‍👦🐏🐳🤳🏽🕵🏿‍♀️', { length: 10 } );
// "emoji 👩‍🍳..."
```

Note: that emoji string would be interpreted as this:
<img width="319" alt="Screenshot 2020-07-16 at 12 57 53" src="https://user-images.githubusercontent.com/2070010/87668518-19b26480-c764-11ea-8798-7c4de480904f.png">

Reply:
I was more thinking about length. Different languages have different sizes for characters, so I was wondering if "pixels" is better than "character length" in these situations. 

To clarify I don't have experience with this personally, so just wondering if there's precedent. Maybe @swissspidy knows.

Reply:
Hmm good question! I think I have to defer to @naokomc to get her view point as a Japanese speaker & locale manager to know whether truncating like this would work in that language.

I know that for word counting we have to differentiate between counting words and characters depending on the locale, but for truncating I am not sure..

Reply:
Quoting [Text size in translation](https://www.w3.org/International/articles/article-text-size.en):

> Chinese, Japanese and Korean, amongst others, are scripts that typically have more complicated characters than those in the Latin script.

I tried to search for the widest Japanese character but couldn't find the answer. AFAIK there isn't a super-wide character in Japanese, they are more or less the same (although, different characters have different kerning if you are using non-proportional font).

Most Japanese character width itself is similar to uppercase "W", which is the widest character in the alphabet.

Reply:
It sounds like this truncate method should work ok then?